### ISSUE TITLE
validation fails with NO_MESSAGES_FILE_IN_LOCALES if directory contains files other than messages.json

### ISSUE THREAD

**Issue by ssborbis** [mozilla/addons-linter#4535]
Place a file other than `messages.json` in a locale directory, i.e. `_locales/en/styles.css` and run the linter

Throws "Empty language directory" error

**../parsers/manfiestjson.js lines 690 - 704**

The first file seen that doesn't match `endsWith("messages.json")` logs an error

```javascript
 if (
          matches &&
          !fileList[i].endsWith(`/${MESSAGES_JSON}`) &&
          // Make sure we do not report the same locale multiple times given
          // that we scan all files in all directories.
          !seen.includes(matches[1])
        ) {
          errors.push(
            messages.noMessagesFileInLocales(
              path.join(LOCALES_DIRECTORY, matches[1])
            )
          );
          seen.push(matches[1]);
        }
      }
```

I haven't seen anywhere stating files other than `messages.json` should not be in `_locale/xx` folders.

**Comment by willdurand** [comment]
I think the problem is that Firefox might throw an error as well in this case, could you verify that by loading an extension temporarily (from `about:debugging`)?

**Comment by ssborbis** [comment]
My extension works fine as a temporary addon, and previous signed versions, but I can't sign / validated to MDN anymore. This problem is relatively new. I was last able to sign through web-ext in late September with no relative changes to my _locale folders.

**Comment by willdurand** [comment]
Oh, so you have `messages.json` and some other files in `_locales/en/`. I guess that's valid although I need to double check why we introduced this change recently.

**Comment by ssborbis** [comment]
This is the troublemaker

https://github.com/mozilla/addons-linter/pull/4505/commits/466d097699e6689e367af1a8dc92fee8ee0e0822

**Comment by willdurand** [comment]
Yeah, and this diff highlights the bug you reported here in the test suite:

```diff
diff --git a/tests/unit/parsers/test.manifestjson.js b/tests/unit/parsers/test.manifestjson.js
index a9bd1276..d8fa75f0 100644
--- a/tests/unit/parsers/test.manifestjson.js
+++ b/tests/unit/parsers/test.manifestjson.js
@@ -3528,6 +3528,7 @@ describe('ManifestJSONParser', () => {
           '_locales/en/messages.json': { size: 1000 },
           '_locales/de/messages.json': { size: 1120 },
           '_locales/hi/messages.json': { size: 1120 },
+          '_locales/hi/unrelated.txt': { size: 0 },
         },
       };
 
```

**Comment by Rob--W** [comment]
Should we really not be emitting anything? I think that we should emit something. There should not be any file other than messages.json in _locales.

Hard-restricting what _locales/ can contain means that we don't have to be concerned with non-localization content in that directory.

**Comment by willdurand** [comment]
Please file a new issue if you want to restrict the files that should live in `_locales/`.

**Comment by Rob--W** [comment]
> Please file a new issue if you want to restrict the files that should live in `_locales/`.

Filed #4540. It's unfortunate that we have shipped a new release(5.19.0) with this change.

**Comment by ioanarusiczki** [comment]
@willdurand On linter 5.19.0 -> with an additional file ( I tested with .js, .txt, .css) -> there is no error message.
